### PR TITLE
Only record general service telemetry on Nexus errors coming from server

### PR DESCRIPTION
**Pull request by pdoerner**
## What changed?

Only calling `telemetryInterceptor.HandleErr` for errors that originated from the server.

## Why?

Other handler errors coming from user workers are expected and should not be logged or included in our service metrics.

## How did you test it?

Existing tests

### CODE REVIEW

File: service/frontend/nexus_handler.go [hunk 1]
```diff
@@ -90,7 +91,7 @@ type operationContext struct {
 	telemetryInterceptor          *interceptor.TelemetryInterceptor
 	redirectionInterceptor        *interceptor.Redirection
 	forwardingEnabledForNamespace dynamicconfig.BoolPropertyFnWithNamespaceFilter
-	cleanupFunctions              []func(error)
+	cleanupFunctions              []func(map[string]string, error)
```
Comment (bergundy) [inline service/frontend/nexus_handler.go:94]: Maybe accept the context here instead? Not critical, we're going to refactor all of this.

File: service/frontend/nexus_handler.go [hunk 2]
```diff
@@ -38,6 +38,9 @@ import (
 	nexuspb "go.temporal.io/api/nexus/v1"
 	"go.temporal.io/api/serviceerror"
 	"go.temporal.io/api/taskqueue/v1"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
```
Comment (bergundy) [inline service/frontend/nexus_handler.go:43]: nit: I see this was already the case but we agreed not to separate temporal from non-temporal imports as a team. If you have some config in your IDE that adds this separation, make sure to remove it.

**Review by bergundy — APPROVED** [review]

File: service/frontend/nexus_handler.go (hunk 1 above)
Comment (pdoerner) [reply service/frontend/nexus_handler.go:94]: I thought about that but figured it was a bit overkill and yeah we're going to change it anyway.

File: service/frontend/nexus_handler.go (hunk 2 above)
Comment (pdoerner) [reply service/frontend/nexus_handler.go]: Yeah I think I removed my IDE settings but it does not automatically combine groups that are separated. Updated this file now.

**Pull request merged**